authentication/authentication.go

```diff
@@ -155,6 +158,7 @@ func New(ctx context.Context, domain string, options ...Option) (*Authentication
 		auth0ClientInfo:   client.DefaultAuth0ClientInfo,
 		idTokenSigningAlg: "RS256",
 		retryStrategy:     client.DefaultRetryOptions,
+		domain:            domain + "/",
```

Comment:
Should we use `url.JoinPath` here instead? If the domain already contains "/"?

Reply:
Yeah that's a good point, on reflection it's probably a bit wasteful to have a separate domain here (just for the added `/`) when we already have url